aten/src/ATen/native/Activation.cpp

```diff
@@ -326,16 +328,37 @@ TORCH_IMPL_FUNC(softshrink_backward_out) (
   shrink_backward_stub(device_type(), *this, lambd);
 }
 
+bool use_mkldnn(const Tensor& input) {
+#if AT_MKLDNN_ENABLED()
+  if (!at::globalContext().userEnabledMkldnn()) {
+    return false;
+  }
+  return (input.is_mkldnn()) || // input is mkldnn Tensor
+    (input.device().is_cpu() &&
+    (((input.scalar_type() == kBFloat16) && mkldnn_bf16_device_check()) ||
+    (input.scalar_type() == kFloat))); // input is dense layout and bfloat16/float32
+#endif
+  return false;
+}
+
 TORCH_IMPL_FUNC(gelu_out_cpu) (
   const Tensor& self, const Tensor& result
 ) {
-  GeluKernel(kCPU, *this);
+  if (use_mkldnn(self)) {
+    at::native::_mkldnn_gelu(self.contiguous(), result);
```

Comment:
I'm a little skeptical about this. Shouldn't you only go to mkldnn if the input is also contiguous; i.e., won't the contiguous() call destroy perf gains for noncontiguous tensors?

Reply:
Now, noncontiguous tensors will go back to the original slow path. When comes to contiguous tensors, this PR uses OneDNN instead of MKL implementation. OneDNN implementation is faster than MKL implementation.

Reply:
OK, so you are saying that if `self` is not contiguous, then `use_mkldnn` will return false. That's fine. But then you don't need to call `contiguous()` here.

Reply:
Yes. Here updated code won't call `contiguous()`. The file is Outdated.